test/python/conftest.py

```diff
@@ -29,96 +34,159 @@ def cleanup():
     yield
 
 
-# ==============================================================================
-
-
+def pattern_to_regex(pattern):
+    no_param = (re.search(r'\[.*\]$', pattern) is None)
+    pattern_noparam = re.sub(r'\[.*$', '', pattern)
+    pattern = re.sub(r'\-', '\\-', pattern)
+    pattern = re.sub(r'\.', '\\.', pattern)
+    pattern = re.sub(r'\[', '\\[', pattern)
+    pattern = re.sub(r'\]', '\\]', pattern)
+    # special case for M.C.F when it possibly macthes with parameterized tests
+    if pattern_noparam.count('.') == 2 and no_param:
+        pattern = '^' + pattern + '\[.*'
+    if pattern_noparam.count('.') == 0:
+        pattern = '^' + pattern + '\..*\..*' + '$'
+    if pattern_noparam.count('.') == 1:
+        pattern = '^' + pattern + '\..*' + '$'
+    return pattern
+
+
+def testfunc_matches_manifest(item, pattern):
+    itemfullname = get_item_fullname(item)  # Module.Class.Func
+    # print('checking for item:', itemfullname, 'with pattern:', pattern)
```

Comment:
remove the commented code if not needed

Reply:
Kept it during the early stage of the code for help with potential bug fixes. But I will remove it.